src/rules/no-descending-specificity/index.js

```diff
@@ -35,6 +35,9 @@ export default function (actual) {
         // Ignore `.selector, { }`
         if (trimSelector === "") { return }
 
+        // Ignore custom property set `--foo: {};`
+        if (/^--.+:$/.test(trimSelector)) { return }
```

Comment:
we have util `isCustomPropertySet` https://github.com/stylelint/stylelint/blob/master/src/utils/isCustomPropertySet.js for this